File: modules/agent_deprecated/src/agent/mocks/repositories/inventory_repo.py
import json
from pathlib import Path

DATA_PATH = Path(__file__).parent.parent / "data"

_stock: list[dict] = []
_locations: list[dict] = []


def _load_data():
    global _stock, _locations
    if _stock:
        return
    with open(DATA_PATH / "inventory.json") as f:
        data = json.load(f)
    _stock = data["stock"]
    _locations = data["locations"]


def get_all_stock() -> list[dict]:
    _load_data()
    return _stock


def find_stock_by_sku(sku: str) -> list[dict]:
    _load_data()
    return [s for s in _stock if s["sku"] == sku]


def find_stock_by_location(location: str) -> list[dict]:
    _load_data()
    return [s for s in _stock if s["location"] == location]


def find_stock_item(sku: str, location: str) -> dict | None:
    _load_data()
    return next((s for s in _stock if s["sku"] == sku and s["location"] == location), None)


def update_stock_item(sku: str, location: str, updates: dict) -> dict | None:
    _load_data()
    item = find_stock_item(sku, location)
    if not item:
        return None
    item.update(updates)
    return item


def append_stock_item(item: dict) -> dict:
    _load_data()
    _stock.append(item)
    return item


def get_all_locations() -> list[dict]:
    _load_data()
    return _locations

File: modules/agent/src/mocks/repositories/inventoryRepo.ts
import { readFileSync } from 'fs';
import { dirname, join } from 'path';
import { fileURLToPath } from 'url';

const __dirname = dirname(fileURLToPath(import.meta.url));
const DATA_PATH = join(__dirname, '..', 'data');

export interface StockItem {
  sku: string;
  product_name: string;
  location: string;
  qty_on_hand: number;
  qty_reserved: number;
  reorder_point: number;
  last_movement: string;
}

export interface Location {
  id: string;
  name: string;
  type: string;
}

let stock: StockItem[] = [];
let locations: Location[] = [];

const loadData = (): void => {
  if (stock.length > 0) return;

  const data = JSON.parse(readFileSync(join(DATA_PATH, 'inventory.json'), 'utf-8'));
  stock = data.stock;
  locations = data.locations;
};

export const getAllStock = (): StockItem[] => {
  loadData();
  return stock;
};

export const findStockBySku = (sku: string): StockItem[] => {
  loadData();
  return stock.filter((s) => s.sku === sku);
};

export const findStockByLocation = (location: string): StockItem[] => {
  loadData();
  return stock.filter((s) => s.location === location);
};

export const findStockItem = (sku: string, location: string): StockItem | undefined => {
  loadData();
  return stock.find((s) => s.sku === sku && s.location === location);
};

export const updateStockItem = (
  sku: string,
  location: string,
  updates: Partial<StockItem>
): StockItem | undefined => {
  loadData();
  const item = findStockItem(sku, location);
  if (!item) return undefined;

  Object.assign(item, updates);
  return item;
};

export const appendStockItem = (item: StockItem): StockItem => {
  loadData();
  stock.push(item);
  return item;
};

export const getAllLocations = (): Location[] => {
  loadData();
  return locations;
};
